server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/node/tasks/CancellableTasksIT.java

```diff
@@ -367,10 +373,28 @@ public void testRemoveBanParentsOnDisconnect() throws Exception {
         } finally {
             allowEntireRequest(rootRequest);
             cancelFuture.actionGet();
-            ensureAllBansRemoved();
+            ensureBansAndCancellationsConsistency();
         }
     }
 
+    public void testChildrenTasksCancelledOnTimeout() throws Exception {
+        Set<DiscoveryNode> nodes = clusterService().state().nodes().stream().collect(Collectors.toSet());
+        final TestRequest rootRequest = generateTestRequest(nodes, 0, between(1, 4), true);
+        ActionFuture<TestResponse> rootTaskFuture = client().execute(TransportTestAction.ACTION, rootRequest);
+        allowEntireRequest(rootRequest);
+        waitForRootTask(rootTaskFuture, true);
+        assertBusy(() -> {
+            for (DiscoveryNode node : nodes) {
+                TransportService transportService = internalCluster().getInstance(TransportService.class, node.getName());
+                for (ThreadPoolStats.Stats stat : transportService.getThreadPool().stats()) {
+                    assertEquals(0, stat.getActive());
```

Comment:
This is a tricky assertion, something might get enqueued just after the assertion. But I think `ensureBansAndCancellationsConsistency` covers most of what we want to assert here.

Reply:
This is necessary because it seems otherwise the test infrastructure may get stuck. If I remember correctly, it's probably because the test at the end then thinks that sometimes things are still running (servers received cancellation but did not make it to process it). That is why I wait for the thread pools to be empty. I do not think anything else can get enqueued afterwards, but if anything does I would expect it to be done by the end of the test procedure.